app/forms/questionnaire_form.py

```diff
@@ -597,3 +617,54 @@ class DynamicForm(QuestionnaireForm):
         data=data,
         formdata=form_data,
     )
+
+
+def _get_block_ids_for_calculated_summary_dependencies(
+    path_finder: PathFinder | None,
+    location: Location | RelationshipLocation | None,
+    schema: QuestionnaireSchema,
+) -> list:
+    # Type ignore: Added to this method as the block will exist at this point
+    if not path_finder:
+        raise ValueError("PathFinder not set")
+
+    if not location:
+        return []
+
+    dependent_sections: set = set()
+    if block_id := location.block_id:
+        if schema.get_block(block_id).get("type") not in [  # type: ignore
+            "ListEditQuestion",
+            "ListAddQuestion",
+            "ListRemoveQuestion",
+            "UnrelatedQuestion",
+            "PrimaryPersonListAddOrEditQuestion",
+        ]:
+            dependent_sections = schema.calculated_summary_section_dependencies_by_block.get(  # type: ignore
+                location.section_id
+            )[
+                block_id
+            ]
+
+    block_dependencies: list = []
+
+    for dependent_section in dependent_sections:
+        if (
+            dependent_section,
+            None,
+        ) in path_finder.progress_store.started_section_keys():
+            path = path_finder.routing_path(
+                section_id=dependent_section, list_item_id=None
+            )
+            block_dependencies.extend(path)
+
+        if (
+            dependent_section,
+            location.list_item_id,
+        ) in path_finder.progress_store.started_section_keys():
+            path = path_finder.routing_path(
+                section_id=dependent_section, list_item_id=location.list_item_id
+            )
+            block_dependencies.extend(path)
```

Comment:
If I remove this it breaks when i'm piping from a non repeating section into a repeating section for my `test_new_calculated_summary_cross_section_dependencies_repeating` as it will only set the dependency with the list_item_id

Reply:
I still don't quite follow. I guess you need to use the list item id if you are currently in a repeat and none if it is not a repeat. A repeating section can't be a dependent unless it is the current repeat.